test/integration/client-side-operations-timeout/client_side_operations_timeout.unit.test.ts

```diff
@@ -93,17 +105,73 @@ describe('CSOT spec unit tests', function () {
   }).skipReason =
     'TODO(NODE-5682): Add CSOT support for socket read/write at the connection layer for CRUD APIs';
 
-  context.skip('Client side encryption', function () {
-    context(
-      'The remaining timeoutMS value should apply to HTTP requests against KMS servers for CSFLE.',
-      () => {}
-    );
+  describe('Client side encryption', function () {
+    describe('KMS requests', function () {
+      const stateMachine = new StateMachine({} as any);
+      const request = {
+        addResponse: _response => {},
+        status: {
+          type: 1,
+          code: 1,
+          message: 'notARealStatus'
+        },
+        bytesNeeded: 500,
+        kmsProvider: 'notRealAgain',
+        endpoint: 'fake',
+        message: Buffer.from('foobar')
+      };
+
+      context('when StateMachine.kmsRequest() is passed a `CSOTimeoutContext`', function () {
+        beforeEach(async function () {
+          sinon.stub(TLSSocket.prototype, 'connect').callsFake(async function (..._args) {
+            await sleep(200);
+            return {} as TLSSocket;
+          });
+        });
+
+        afterEach(async function () {
+          sinon.restore();
+        });
+
+        it('the kms request times out through remainingTimeMS', async function () {
+          const timeoutContext = new CSOTTimeoutContext({
+            timeoutMS: 500,
+            serverSelectionTimeoutMS: 30000
+          });
+          sleep(300);
+          const err = await stateMachine.kmsRequest(request, timeoutContext).catch(e => e);
+          expect(err).to.be.instanceOf(MongoOperationTimeoutError);
+          expect(err.errmsg).to.equal('KMS request timed out');
+        });
+      });
+
+      context('when StateMachine.kmsRequest() is not passed a `CSOTimeoutContext`', function () {
+        beforeEach(async function () {
+          sinon.stub(TLSSocket.prototype, 'connect').callsFake(async function (..._args) {
+            return {} as TLSSocket;
+          });
+        });
+
+        afterEach(async function () {
+          sinon.restore();
+        });
+
+        it('the kms request does not timeout within 30 seconds', async function () {
+          const sleepingFn = async () => {
+            await sleep(55000);
```

Comment:
```suggestion
            await sleep(30000);
```

Also - Node has a setTimeout that returns a promise we can use now, if you import it from `timers/promises`